File: labs/D80_Builder/python/complete/house.py
from floor import Floor
from foundation import Foundation
from roof import Roof
from wall import Wall
from typing import List

class House():
    def __init__(self, foundation: Foundation, floors: List[Floor], walls: List[Wall], roof: Roof):
        self.roof = roof
        self.walls = walls
        self.floors = floors
        self.foundation = foundation

    def print_summary(self):
        print('House Summary')
        print('Foundation: ' + self.foundation.material)
        print('Walls:')
        for wall in self.walls:
            print('  ' + wall.material)
        print('Floors:')
        for floor in self.floors:
            print('  ' + floor.material)
        print('Roof: ' + self.roof.material)

File: labs/D80_Builder/java/startingpoint/app/src/main/java/BuilderStartingpoint/House.java
package BuilderStartingpoint;

import java.util.Arrays;

public class House {
    private Roof roof;
    private Foundation foundation;
    private Wall[] walls;
    private Floor[] floors;

    public Roof getRoof() {
        return roof;
    }

    public void setRoof(Roof roof) {
        this.roof = roof;
    }

    public Foundation getFoundation() {
        return foundation;
    }

    public void setFoundation(Foundation foundation) {
        this.foundation = foundation;
    }



    public House(Roof roof, Foundation foundation, Wall[] walls, Floor[] floors) {
        this.roof = roof;
        this.foundation = foundation;
        this.walls = walls;
        this.floors = floors;
    }

    @Override
    public String toString() {
        return "House \n[floors=" + Arrays.toString(floors) + ", \nfoundation=" + foundation + ", \nroof=" + roof + ", \nwalls="
                + Arrays.toString(walls) + "\n]";
    }

  
}
